# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../hugin/bad_openexr_hack.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur hugin-0.8.0-orig/CMakeModules/FindOpenEXR.cmake hugin-0.8.0/CMakeModules/FindOpenEXR.cmake
--- hugin-0.8.0-orig/CMakeModules/FindOpenEXR.cmake	2009-07-22 20:23:20.000000000 +0100
+++ hugin-0.8.0/CMakeModules/FindOpenEXR.cmake	2009-07-22 20:26:14.000000000 +0100
@@ -22,6 +22,12 @@
   INCLUDE(UsePkgConfig)
   
   PKGCONFIG(OpenEXR _OpenEXRIncDir _OpenEXRLinkDir _OpenEXRLinkFlags _OpenEXRCflags)
+
+  # ugly hack, sets BAD_OPENEXR varible when OpenEXR <= 1.5.0 is used
+  # because these old versions doesn't set -lpthread flag correcly
+  pkg_check_modules(OPENEXR OpenEXR)
+  STRING(COMPARE GREATER OPENEXR__VERSION 1.5.0 BAD_OPENEXR)
+
 ENDIF (NOT WIN32 OR MINGW)
   FIND_PATH(OPENEXR_INCLUDE_DIR ImfRgbaFile.h
      ${_OpenEXRIncDir}
@@ -134,7 +140,11 @@
        else (MSVC)
          set(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} ${OPENEXR_ILMTHREAD_LIBRARY})
        endif (MSVC)
-     else (OPENEXR_ILMTHREAD_LIBRARY)
+# if BAD_OPENEXR is set, add -lpthread flag manually
+       if (BAD_OPENEXR)
+         set(OPENEXR_LIBRARIES ${OPENEXR_LIBRARIES} -lpthread)
+       endif (BAD_OPENEXR)
+	else (OPENEXR_ILMTHREAD_LIBRARY)
        if (MSVC)
          set(OPENEXR_LIBRARIES optimized ${OPENEXR_IMATH_LIBRARY} optimized ${OPENEXR_ILMIMF_LIBRARY} optimized ${OPENEXR_IEX_LIBRARY} optimized ${OPENEXR_HALF_LIBRARY}  debug ${OPENEXR_IMATH_LIBRARY_DEBUG} debug ${OPENEXR_ILMIMF_LIBRARY_DEBUG} debug ${OPENEXR_IEX_LIBRARY_DEBUG} debug ${OPENEXR_HALF_LIBRARY_DEBUG} CACHE STRING "The libraries needed to use OpenEXR")
        else (MSVC)
